libbeat/processors/add_cloud_metadata/provider_aws_ec2.go

```diff
@@ -94,8 +95,20 @@ func fetchRawProviderMetadata(
 ) {
 	logger := logp.NewLogger("add_cloud_metadata")
 
+	// generate AWS specific client with overriding requirements
+	var awsHTTPClient awshttp.BuildableClient
+	awsHTTPClient = *awsHTTPClient.WithTimeout(client.Timeout)
+	awsHTTPClient = *awsHTTPClient.WithTransportOptions(func(tr *http.Transport) {
+		transport, ok := client.Transport.(*http.Transport)
+		if ok {
+			tr.TLSClientConfig = transport.TLSClientConfig
+		}
+
+		tr.DisableCompression = true
```

Comment:
Why are we setting `tr.DisableCompression = true`?

Reply:
ahh sorry, already updated this :) should have been `DisableKeepAlives` and matches with generic client. Also related to this discussion https://github.com/elastic/beats/pull/44189#discussion_r2073936702